src/miral/basic_window_manager.cpp

```diff
@@ -2881,6 +2881,23 @@ void miral::BasicWindowManager::update_application_zones_and_attached_windows()
             }
         }
 
+        // We are sorting east/west connected edges to the back of the list so that they
+        // force everyone else out of the way when they're placed.
+        std::sort(first_pass.begin(), first_pass.end(), [](WindowInfo* first, WindowInfo* second) {
```

Comment:
1. Wouldn't `partition` (or `remove_if`!) be more efficient than `sort`?
2. Maybe we should split `first_pass` into a `horizontal` and `vertical` pass? instead of partitioning?

Reply:
2 sounds good!